File: internal/api/middleware/logger.go
package middleware

import (
	"time"

	"github.com/gofiber/fiber/v2"
	"github.com/rs/zerolog"
	"github.com/rs/zerolog/log"
)

// LoggerConfig cria middleware personalizado para logging de requisições
func LoggerConfig() fiber.Handler {
	return func(c *fiber.Ctx) error {
		start := time.Now()

		// Processar a requisição
		err := c.Next()

		// Calcular duração
		duration := time.Since(start)

		// Obter informações da requisição
		method := c.Method()
		path := c.Path()
		statusCode := c.Response().StatusCode()
		ip := c.IP()
		userAgent := c.Get("User-Agent")

		// Determinar nível de log baseado no status code
		var event *zerolog.Event
		switch {
		case statusCode >= 500:
			event = log.Error()
		case statusCode >= 400:
			event = log.Warn()
		default:
			event = log.Info()
		}

		// Registrar log estruturado
		event.
			Str("method", method).
			Str("path", path).
			Int("status", statusCode).
			Dur("duration", duration).
			Str("ip", ip).
			Str("user_agent", userAgent).
			Msgf("%s %s - %d (%s)", method, path, statusCode, duration)

		return err
	}
}

File: backup/nodejs-original/src/middleware/logger.js
/**
 * Request logging middleware
 */
const logger = (req, res, next) => {
    const start = Date.now();
    
    // Log request
    console.log(`${new Date().toISOString()} - ${req.method} ${req.path} - IP: ${req.ip} - User-Agent: ${req.get('User-Agent') || 'Unknown'}`);
    
    // Override res.end to log response
    const originalEnd = res.end;
    res.end = function(chunk, encoding) {
        const duration = Date.now() - start;
        console.log(`${new Date().toISOString()} - ${req.method} ${req.path} - ${res.statusCode} - ${duration}ms`);
        
        // Call original end method
        originalEnd.call(this, chunk, encoding);
    };
    
    next();
};

module.exports = logger;